ui/src/app/settings/hooks.ts

```diff
@@ -47,6 +55,13 @@ export const useDhcpTarget = (
     (!!subnetId && subnetLoaded) ||
     (!!nodeId && controllerLoaded && deviceLoaded && machineLoaded);
 
+  useEffect(() => {
+    dispatch(subnetActions.fetch());
+    dispatch(controllerActions.fetch());
+    dispatch(deviceActions.fetch());
+    dispatch(machineActions.fetch());
+  }, [dispatch]);
```

Comment:
I'm of two minds about this general pattern of always dispatching fetch actions. Most of the time its fine when its handled by a larger parent component, but in a hook like this that's used in table rows these actions are going to be fetched a lot. Sure the websocket saga prevents the fetch actually happening, but it's a lot of unnecessary noise that's especially unhelpful when going over the action logs.

All this to say this is fine to land because it's a pattern used pretty much everywhere, but I'd like it if we came up with an alternative. Maybe a simple hook for each model would be enough...
``` typescript
export const useFetchSubnets = (): void => {
  const dispatch = useDispatch();
  const subnetsLoaded = useSelector(subnetSelectors.loaded);

  useEffect(() => {
    if (!subnetsLoaded) {
      dispatch(subnetActions.fetch());
    }
  }, [dispatch, subnetsLoaded]);
};
```

Reply:
Totally agree with you. Checking if that has been loaded before fetching did cross my mind, but decided no to do it for the reason you mentioned.

> Maybe a simple hook for each model 

I like your thinking. I was thinking about something similar a while ago... What if we provided something like useData/useQuery type-of hook for each model? We could potentially make quite a few improvements as this new layer of abstraction wouldn't be so coupled to the redux store (e.g. including a single status string as an option).

```javascript
const { status, loading, loaded (...), data, error } = useQuery('subnets')
``````

We could even make something that would allow for a combination of models:
```javascript
const [ subnets, devices ] = useQueries(["subnets", "devices"])

subnets.status === "success" ? subnets.data.map(...
```

...or add a single status / allLoaded or similar and not have to duplicate this logic in all components that need data from multiple models like we do now.
```javascript
const { status, slices: [subnets, devices] ] = useQueries(["subnets", "devices"])
```

Reply:
https://github.com/canonical-web-and-design/maas-ui/issues/3896